File: c/Basic.c
#include <stdio.h> //printf, scanf, fgets
#include <math.h> //pow, sqrt, floor, ceil

// container main() is executed first
int main(){ 
    // show a text on screen
    printf("Hello world!\n");
    
    // draw a triangle
    printf("\n");
    printf("   /|\n");
    printf("  / |\n");    
    printf(" /  |\n");
    printf("/___|\n");   

    // using string and integer variable
    char charName[] = "John Doe";
    int charAge = 27;
    printf("There once was a man named %s\n", charName);
    printf("he was %d years old.\n", charAge);
    printf("%s really liked his/her name, but did not like being %d.\n", charName, charAge);

    // using integer manipulation
    charAge = 30;
    printf("%s is %d years old now.\n", charName, charAge);

    // using placeholder with direct values
    printf("%s age is %d.\n", "Mark Wahlberg", 38);

    // double precision floating point will be rounded
    double phi = 3.14159265358979323846;
    printf("%s is %f.\n", "Phi", phi);

    // you can do float operation into double precision
    printf("%f\n", 5 / 0.5);
    int twelve = 12;
    float half = 0.5;
    printf("%f\n", twelve * half);

    // using power function (exponent)
    printf("%f\n", pow(2, 3));

    // square root
    printf("%f\n", sqrt(81));

    // ceiling and floor
    printf("%f and %f\n", ceil(24.5), floor (24.5));

    /*
    comment block
    */

    // constant/unchangeable var.
    const int EIGHT = 8;
    printf("this number is constant: %d.\n", EIGHT);

    // accepts user input as int
    int age;
    printf("Enter your year of birth: ");
    scanf("%d", &age);
    printf("%d is your year of birth.\n", age);

    // simple addition calculator
    double num1, num2;
    printf("Enter first number: ");
    scanf("%lf", &num1);
    printf("Enter second number: ");
    scanf("%lf", &num2);
    printf("%f + %f = %f\n", num1, num2, num1 + num2);

    // stops the runs with exit code 0 (0x0 which mean a success)
    return 0;
}

File: php/basic.php
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Learn the Basics!</title>
</head>
<body>
    <!-- PHP Section -->
    <?php
        // using inline HTML
        echo "<h1>My tiny PHP site</h1>";
        echo "<hr>";
        echo "<p>First paragraph comes here.</p><br>";

        // declaring a variables
        $characterName = "John Doe";
        $characterAge = 27;
        
        echo "There once was a man named $characterName <br>";
        echo "He was $characterAge years old. <br>";

        // changing the same variable midway
        $characterName = "Mike Burgowski";
        $characterAge = 40;

        echo "He really liked the name $characterName <br>";
        echo "But didn't like being $characterAge years old. <br>";

        // declaring more types of vars.
        $myBool = false;
        $myGpa = 3.25;
        $myString = "hehe";

        // display value directly from a variable
        echo "<br>";
        echo $myGpa; echo "<br>";

        // display string in lower and uppercase function
        echo "<br>";
        $mixedCase = "hEhEeHeeHEh";
        echo $mixedCase; echo "<br>";
        echo "uppercased: ",strtoupper($mixedCase); echo "<br>";
        echo "lowercased: ",strtolower($mixedCase); echo "<br>";
        echo strtolower("This Is A Word"); echo "<br>";
        
        // display the char. length of a string variable
        echo "length: ",strlen($mixedCase); echo "<br>";

        // get character index in variable
        $i = 1;
        echo "char. index ",$i," is: ",$mixedCase[$i]; echo "<br>";
    
        // change first letter to B
        $mixedCase[0] = "B";
        echo $mixedCase; echo "<br><br>";

        // find and replace certain characters function
        $univName = "Universitas Indonesia";
        echo str_replace("Universitas", "University of", $univName); echo "<br><br>";
    
        // using substring function, 12 is the starting point index
        echo substr($univName, 12); echo "<br>";
        // grab 4 characters length, starting from index 12
        echo substr($univName, 12, 4); echo "<br><br>";
    
        // will printout the answer directly
        echo 5 + 9; echo "<br>";
        echo 5.7 * 9; echo "<br>";
        echo 10 % 3; echo "<br><br>";

        // PHP uses PEMDAS order of operations
        echo 4 + 10 * 5 / 2 - 3; echo "<br><br>";

        // integer variable manipulation
        $aNumber = 9;
        $aNumber++;
        echo $aNumber; echo "<br>";
        $aNumber - 5;
        echo $aNumber; echo "<br><br>";

        // using absolute function
        echo abs(-19); echo "<br><br>";

        // using power
        echo pow(2, 3); echo "<br><br>";

        // using square root
        echo sqrt(100); echo "<br><br>";

        // using round, floor, and ceil
        $myFloatNum = 3.5;
        echo "original number: $myFloatNum <br>";
        echo round($myFloatNum),", ",floor($myFloatNum),", ",ceil($myFloatNum); echo "<br><br>";
    ?>
</body>
</html>